.github/workflows/deploy-image.yaml

```diff
@@ -0,0 +1,106 @@
+#        required: true
+#        default: bullseye
+#      imageVariant:
+#        description: Base image variant
+#        required: true
+#        default: slim
+  workflow_run:
+    workflows:
+      - build-latest
+    branches:
+      - develop
+    types:
+      - completed
+jobs:
+  deploy-image:
+    runs-on: ubuntu-latest
+    env:
+      latest-ref: refs/heads/develop
+    strategy:
+      matrix:
+        postgresMajorVersion:
+          - 13
+        postgisMajorVersion:
+          - 3
+        postgisMinorRelease:
+          - 1
+        include:
+          - distro: debian
+            imageVersion: bullseye
+            imageVariant: slim
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Get build cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          # Build inputs are:
+          # - Dockerfile recipe
+          # - docker-compose.build.yml
+          # - build args (.example.env)
+          # - base_build directory
+          # - scripts directory
+          key: buildx-${{ hashFiles('Dockerfile', 'docker-compose.build.yml', '.example.env', 'base_build', 'scripts') }}-${{ github.sha }}
+          restore-keys: |
+            buildx-${{ hashFiles('Dockerfile', 'docker-compose.build.yml', '.example.env', 'base_build', 'scripts') }}-
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Build base image
+        id: docker_build_base
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKERHUB_REPO }}/postgis:base-${{ matrix.distro }}-${{ matrix.imageVersion }}-${{ matrix.imageVariant }}
+          build-args: |
+            DISTRO=${{ matrix.distro }}
+            IMAGE_VERSION=${{ matrix.imageVersion }}
+            IMAGE_VARIANT=${{ matrix.imageVariant }}
+            GENERATE_ALL_LOCALE=1
+            POSTGRES_MAJOR_VERSION=${{ matrix.postgresMajorVersion }}
+            POSTGIS_MAJOR_VERSION=${{ matrix.postgisMajorVersion }}
+            POSTGIS_MINOR_VERSION=${{ matrix.postgresMinorVersion }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          target: postgis-base
+
+      - name: Build prod image
+        id: docker_build_prod
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile
+          push: true
+          load: true
+          tags: ${{ secrets.DOCKERHUB_REPO }}/postgis:${{ matrix.postgresMajorVersion }}-${{ matrix.postgisMajorVersion }}.${{ matrix.postgisMinorRelease }}
```

Comment:
RFC to @NyakudyaA .

I propose we should also change the tag from `13.1` into `13-3.1` where the first part before dash is the postgres major version, and the rest is postgis full version to reduce ambiguity. 

Ideally we should also define the orchestration version somewhere in the back, like `13-3.1-21.01` where `21.01` is a shorthand for 2021 January. I prefer using `21.01` for orchestration tag (to version the Dockerfile recipe or build scripts) instead of semver because we don't yet have any clear convention to define a 'feature'.

The specific tag can be used for production usage (to really pin down the image), like `13-3.1-21.01.19`. But we should also track the common uses tag like `13` or `13.1` where user can easily type it for testing.

In practice, the common tag will always move. Today it can be that `13` is the same image as `13.1`, `13-3.1`, or `13-3.1-21.01` (in the order of specificity). Next month it could be that it points to `13-3.1-21-02`.

The one who will keep the tags to be up to date is the docker-meta actions (not yet being put here).

Reply:
I am ok with this. The convention was always that we always had one or two minor releases in each cycle. 

Something related to this versioning. Do we have a way to track if there is a new minor release so that we can automatically publish the corresponding image. Maybe we could investigate if we cannot do so for now

Reply:
@NyakudyaA , did you mean tracking new minor release from upstream? (like from Postgis source repo?). If this is the case, we can only do that via dependency bot.

Reply:
Yes, But we can do it in another PR